docker/dockerfiles/dev-env/dev-env.Dockerfile

```diff
@@ -68,3 +68,16 @@ COPY --from=builder $STARROCKS_THIRDPARTY $STARROCKS_THIRDPARTY
 COPY --from=builder /root/.m2 /root/.m2
 # Copy starlet dependencies
 COPY --from=starlet /release $STARLET_INSTALL_DIR
+# remove unnecessary and big starlet dependencies
+RUN rm -r $STARLET_INSTALL_DIR/third_party/include/rocksdb
+RUN rm -r $STARLET_INSTALL_DIR/third_party/include/brpc
+RUN rm -r $STARLET_INSTALL_DIR/third_party/include/aws
+RUN rm -r $STARLET_INSTALL_DIR/third_party/include/boost
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/librocksdb.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libbrpc.a
+RUN rm -r $STARLET_INSTALL_DIR/third_party/lib/aws*
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libaws*.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libboost*.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libprotobuf*.a
+RUN rm $STARLET_INSTALL_DIR/third_party/lib/libprotoc.a
+RUN rm $STARLET_INSTALL_DIR/third_party/bin/protoc*
```

Comment:
make the deleted items into a file `starlet_redundant_lists.txt`

```
RUN for line in `cat starlet_redundant_lists.txt` ; do rm $STARLET_INSTALL_DIR/$line ; done
```

Also need to create a separate stage to perform the delete operations, and then copy the remainder to $STARLET_INSTALL_DIR